### ISSUE TITLE
Requiring non-pyproject.toml dependency from git fails

### ISSUE THREAD

**Issue by YannickJadoul** [pypa/build#242]
Having this as minimal `pyproject.toml` seems to fail since `build` 0.2.0, while it was working before in 0.1.0:

```toml
[build-system]
requires = [
    "setuptools",
    "scikit-build @ git+https://github.com/scikit-build/scikit-build",
]
build-backend = "setuptools.build_meta"
```

(Yes, this seems silly if there are wheels on PyPI, but in practice, I am using the installation from GitHub since I have a few changes in my own fork, pending to hopefully be merged, and not part of a released.)

Running this results in the following output and error. It seems `setuptools` is required in the isolated environment?

```
$ python3 -m build
Collecting scikit-build@ git+https://github.com/scikit-build/scikit-build
  Cloning https://github.com/scikit-build/scikit-build to /tmp/pip-install-s5r9t_4x/scikit-build_1f68c67eb9c24223ae2c9bdd421cb361
  Running command git clone -q https://github.com/scikit-build/scikit-build /tmp/pip-install-s5r9t_4x/scikit-build_1f68c67eb9c24223ae2c9bdd421cb361
    ERROR: Command errored out with exit status 1:
     command: /tmp/build-env-g7qinbi_/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-s5r9t_4x/scikit-build_1f68c67eb9c24223ae2c9bdd421cb361/setup.py'"'"'; __file__='"'"'/tmp/pip-install-s5r9t_4x/scikit-build_1f68c67eb9c24223ae2c9bdd421cb361/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-_ds_qxfy
         cwd: /tmp/pip-install-s5r9t_4x/scikit-build_1f68c67eb9c24223ae2c9bdd421cb361/
    Complete output (3 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    ModuleNotFoundError: No module named 'setuptools'
    ----------------------------------------
WARNING: Discarding git+https://github.com/scikit-build/scikit-build. Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement scikit-build (unavailable)
ERROR: No matching distribution found for scikit-build (unavailable)
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/yannick/.local/lib/python3.6/site-packages/build/__main__.py", line 213, in <module>
    main(sys.argv[1:], 'python -m build')
  File "/home/yannick/.local/lib/python3.6/site-packages/build/__main__.py", line 205, in main
    build_package(args.srcdir, outdir, distributions, config_settings, not args.no_isolation, args.skip_dependencies)
  File "/home/yannick/.local/lib/python3.6/site-packages/build/__main__.py", line 94, in build_package
    _build_in_isolated_env(builder, outdir, distributions, config_settings)
  File "/home/yannick/.local/lib/python3.6/site-packages/build/__main__.py", line 56, in _build_in_isolated_env
    env.install(builder.build_dependencies)
  File "/home/yannick/.local/lib/python3.6/site-packages/build/env.py", line 142, in install
    subprocess.check_call(cmd)
  File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/tmp/build-env-g7qinbi_/bin/python', '-Im', 'pip', 'install', '--no-warn-script-location', '-r', '/tmp/build-reqs-u2w_rr7_.txt']' returned non-zero exit status 1.
```

Is this an error to be fixed in `scikit-build` or is this an unintended regression wrt. 0.1.0?

**Comment by henryiii** [comment]
I think this was fixed in #232.

**Comment by YannickJadoul** [comment]
Yes, that makes things work indeed! Sorry, I should have tried that.

Then I'll just wait for the next release to have things fixed :-)

**Comment by henryiii** [comment]
Fixed in 0.3.0!

**Comment by YannickJadoul** [comment]
Thanks! Things seem to be working again, indeed :-)
